### DISCUSSION TITLE
Redirecting ouptut to console instead of repl

### DISCUSSION THREAD

**Discussion by asmodeus812** [mfussenegger/nvim-dap#631] [Q&A]
Is it possible to redirect output to the regular console instead of repl. At the moment most of my adpaters - node2, coreclr, php, markdown, go, dart, python and others, go to the repl when you output something with print/consolewriteln etc. in the respective languaes. For adapters like lldb, cppdbg and lldb-vscode using the terminal "integrated" options seems to work and the output of those goes directly into the console. Java as works fine as well, using integratedTerminal. 
Tried using console = "integratedTerminal" with the ones mentioned above but output still goes to repl instead. Note: i am using nvim-dap-ui

**Comment by mfussenegger** [comment]
I'm confused, nvim-dap by default has the repl and the integrated terminal. What is the regular console?

**Reply by asmodeus812** [reply]
Yeah, i was referring to the integrated terminal @mfussenegger. But setting to integrated terminal does not seem to work for all adapters, as i mentiond above some are still displaying stdout to the repl.

**Reply by mfussenegger** [reply]
I see.
By default the integrated terminal shows the output of your application. The debug adapter itself is allowed to send additional output via a output event. Currently these debug adapter outputs end up in the repl and there is no way to change that.

I could add some configuration to change that, but not sure if it is warranted.

What kind of output are you seeing in the repl?

**Reply by asmodeus812** [reply]
Every output that gets printed by the application, the usual stdout, goes in the repl instead - for example dart's print() statements or same goes for node, go, python etc. That should usually go to the integratedTerminal, as it does for java or any of the 3 available debuggers for c/cpp/rust for example.

**Reply by stevenhay** [reply]
Hi @mfussenegger I hope you're doing well. I'm also interested in this as I'm facing the exact same issue. For a little more info, when using Go with dlv, all program output, not just "additional output" goes directly to the repl instead of the integrated terminal.
![image](https://user-images.githubusercontent.com/26744309/197315690-0c12ead7-9ab9-4cf1-bfbf-343713c89761.png)

In this image you can see the standard program output using `fmt.Println` and `fmt.Printf` going directly to the repl instead of the integrated terminal. As the OP said, some adapters like jdtls output their program output correctly to the integrated terminal.
